docs/deployment/pull-backup.rst

```diff
@@ -174,3 +188,116 @@ directly extract it without the need of mounting with SSHFS:
 Note that in this scenario the tar format is the limiting factor – it cannot
 restore all the advanced features that BorgBackup supports. See
 :ref:`borg_export-tar` for limitations.
+
+socat
+=====
+
+In this setup a SSH connection from the backup server to the client is
+established that uses SSH reverse port forwarding to transparently
+tunnel data between UNIX domain sockets on the client and server and the socat
+tool to connect these with the borg client and server processes, respectively.
+
+The program socat has to be available on the backup server and on the client
+to be backed up.
+
+When **pushing** a backup the borg client (holding the data to be backed up)
+connects to the backup server via ssh, starts ``borg serve`` on the backup
+server and communicates via standard input and output (transported via SSH)
+with the process on the backup server.
+
+With the help of socat this process can be reversed. The backup server will
+create a connection to the client (holding the data to be backed up) and will
+**pull** the data.
+
+In the following example *borg-server* connects to *borg-client* to pull a backup.
+
+To provide a secure setup sockets should be stored in ``/run/borg``, only
+accessible to the users that run the backup process. So both systems,
+*borg-server* and *borg-client* the folder ``/run/borg`` has to be created::
+
+   sudo mkdir -m 0700 /run/borg
+
+On *borg-server* the socket file is opened by the user accessing
+``/path/to/repo``, so the user has to have read and write permissions on
+``/run/borg``::
+
+   borg-server:~$ sudo chown user_accessing_path_to_repo /run/borg
```

Comment:
another idea instead of always repeating this long `user_accessing_path_to_repo` (and similar, or long paraphrasings of this) would be to define early and then always use:

- `borgs` as the username running `borg serve` (on the server)
- `borgc` as the username running `borg` (on the client)

Reply:
(and also note that this is used for clarification and other user names might be required / can be working also, as long as one is consistent)

Reply:
I've added e5ef137eccb6267797e98b4b71a29f1ecc7e7963 for this